akka-projection-samples/src/main/scala/akka/projection/Samples.scala

```diff
@@ -0,0 +1,168 @@
+package akka.projection
+
+import akka.{ Done, NotUsed }
+import akka.actor.ActorSystem
+import akka.kafka.{ CommitterSettings, ConsumerSettings, Subscriptions }
+import akka.kafka.scaladsl.{ Committer, Consumer }
+import akka.persistence.cassandra.query.scaladsl.CassandraReadJournal
+import akka.persistence.jdbc.query.scaladsl.JdbcReadJournal
+import akka.persistence.query.scaladsl.EventsByTagQuery
+import akka.persistence.query.{ EventEnvelope, Offset, PersistenceQuery, TimeBasedUUID }
+import akka.projection.KafkaAtLeastOnceSample.projection
+import akka.projection.KafkaSourceWithTransactionalJDBCSample.{ projection, system }
+import akka.projection.scaladsl.jdbc.{ JdbcOffsetStore, JdbcOffsetStoreAkkaOffset, JdbcOffsetStoreLong, JdbcProjection }
+import akka.projection.scaladsl.kafka.{ KafkaConsumer, KafkaProjection }
+import akka.stream.scaladsl.Source
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.TopicPartition
+
+import scala.concurrent.Future
+
+/**
+ * A Kafka projection that commits the offset in the topic using the Committer.flow (at-least-once)
+ */
+object KafkaAtLeastOnceSample {
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  val consumerSettings: ConsumerSettings[String, String] = ???
+  val committerSettings = CommitterSettings(system).withMaxBatch(100)
+
+  val src = KafkaConsumer.atLeastOnce(
+    consumerSettings.withGroupId("group1"),
+    Subscriptions.topics("topic1"),
+    committerSettings) { msg =>
+    Future.successful("pretending that this is projected to somewhere")
+  }
+
+  // a pure Kafka projection is super simple, as it only requires a Source
+  val projection = KafkaProjection(src)
+  projection.start()
+  projection.stop()
+}
+
+object KafkaAtLeastOnceSample2 {
+  implicit val system = ActorSystem("sample")
+  implicit val ec = system.dispatcher
+
+  val consumerSettings: ConsumerSettings[String, String] = ???
+  val committerSettings = CommitterSettings(system).withMaxBatch(100)
+
+  val src = Consumer
+    .committableSource(consumerSettings, Subscriptions.topics("topic1"))
+    .mapAsync(1) { msg =>
+      Future.successful(s"pretending that this is projected to somewhere").map(_ => msg.committableOffset)
```

Comment:
It would make it easier if the samples projected the data somewhere. Counting elements might be enough. Updating some persistent model even more concrete.

Reply:
Maybe [DynamoDB](https://doc.akka.io/docs/alpakka/2.0/dynamodb.html#sending-requests-and-receiving-responses) or [ElasticSearch](https://doc.akka.io/docs/alpakka/2.0/elasticsearch.html#elasticsearch-as-flow)

Reply:
The intention of the samples was to show how the APIs would look like. I don't think we will make it more evolved that it already is.